src/core_plugins/elasticsearch/lib/check_es_version.js

```diff
@@ -1,31 +1,70 @@
+/**
+ * ES and Kibana versions are locked, so Kibana should require that ES has the same version as
+ * that defined in Kibana's package.json.
+ */
+
 import _ from 'lodash';
 import esBool from './es_bool';
-import versionSatisfies from './version_satisfies';
+import parseVersionNumbers from './parse_version_numbers';
+import isEsCompatibleWithKibana from './is_es_compatible_with_kibana';
+import isEsNewerThanKibana from './is_es_newer_than_kibana';
 import SetupError from './setup_error';
 
-module.exports = function (server) {
+module.exports = function checkEsVersion(server, kibanaVersion) {
   server.log(['plugin', 'debug'], 'Checking Elasticsearch version');
 
   const client = server.plugins.elasticsearch.client;
-  const engineVersion = server.config().get('elasticsearch.engineVersion');
 
   return client.nodes.info()
   .then(function (info) {
-    const badNodes = _.filter(info.nodes, function (node) {
-      // remove nodes that satify required engine version
-      return !versionSatisfies(node.version, engineVersion);
-    });
+    // Aggregate incompatible ES nodes.
+    const incompatibleNodes = [];
+
+    // Aggregate ES nodes which should prompt a Kibana upgrade.
+    const warningNodes = [];
 
-    if (!badNodes.length) return true;
+    _.forEach(info.nodes, esNode => {
+      if (!isEsCompatibleWithKibana(esNode.version, kibanaVersion)) {
+        return incompatibleNodes.push(esNode);
+      }
 
-    const badNodeNames = badNodes.map(function (node) {
-      return 'Elasticsearch v' + node.version + ' @ ' + node.http_address + ' (' + node.ip + ')';
+      // It's acceptable if ES is ahead of Kibana, but we want to prompt users to upgrade Kibana
+      // to match it.
+      if (isEsNewerThanKibana(esNode.version, kibanaVersion)) {
+        warningNodes.push(esNode);
+      }
     });
 
-    const message = `This version of Kibana requires Elasticsearch ` +
-    `${engineVersion} on all nodes. I found ` +
-    `the following incompatible nodes in your cluster: ${badNodeNames.join(',')}`;
+    function getEsNodeNames(nodes) {
+      return nodes.map(node => {
+        return 'Elasticsearch v' + node.version + ' @ ' + node.http_address + ' (' + node.ip + ')';
+      });
+    }
+
+    if (warningNodes.length) {
+      const warningNodeNames = getEsNodeNames(warningNodes);
+
+      const warningMessage =
+        `This version of Kibana (${kibanaVersion}) is behind some of the versions of Elasticsearch ` +
+        `on the nodes in your cluster. Kibana is behind the following nodes in your cluster: ` +
+        `${warningNodeNames.join(',')}`;
+
+      server.log(['warning'], {
+        tmpl: warningMessage,
```

Comment:
Implemented to follow this structure:
```json
{  
   "type":"log",
   "@timestamp":"2016-09-13T18:14:52Z",
   "tags":[  
      "warning"
   ],
   "pid":45003,
   "kibanaVersion":"6.0.0-alpha1",
   "nodes":[  
      {  
         "version":"6.0.0-alpha1",
         "http":{  
            "publish_address":"127.0.0.1:9200"
         },
         "ip":"127.0.0.1"
      }
   ],
   "message":"You're running Kibana 6.0.0-alpha1 with some newer versions of Elasticsearch. Update Kibana to the latest version to prevent compatibility issues: v6.0.0-alpha1 @ 127.0.0.1:9200 (127.0.0.1)"
}
```